### DISCUSSION TITLE
Get extension recommended "based on recently opened files"

### DISCUSSION THREAD

**Discussion by BeFunes** [microsoft/vscode-discussions#170] [Extension Development QnA] [answered]
Hello! 

I'd like to get my extension recommended "Based on recently opened files".
This option is mentioned in the docs [here](https://code.visualstudio.com/docs/editor/extension-marketplace#_recommended-extensions), and I've seen other extensions doing this.

However I can't seem to find any documentation on how achieve this.

Does anyone know how I can do this?

**Comment by gjsjohnmurray** [comment, marked as answer]
This features seems to be implemented in https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/extensions/browser/fileBasedRecommendations.ts

Pinging @sandy081 for more information.

**Reply by sandy081** [reply]
What is your extension and how do you want it to get recommended?

CC @isidorn @digitarald

**Reply by BeFunes** [reply]
Hey! 
My extension is not out yet, so this question is perhaps a little premature. 
It will be the official CircleCI extension (authored by CircleCI) - and provides language support for CircleCI configuration files. 
We'd love to have it recommended when users open `.yml` or `.yaml` files contained inside a `.circleci/` directory.

**Reply by isidorn** [reply]
Once your extension is released and out in the open for at least a couple of months let us know.
We usually do not recommend extensions based on file types, but we can help with social media outreach.
Once you publish your extension I suggest you send me an email inikolic@microsoft.com so we can help with promotion. Thanks

**Reply by BeFunes** [reply]
Hi @sandy081, the extension has now been published: [CircleCI](https://marketplace.visualstudio.com/items?itemName=circleci.circleci). It's the official CircleCI VS Code extension, so we think it would be really helpful to recommend it to help users navigate through CircleCI-specific YAML syntax. 
I've exchanged with @digitarald on the slack community about this a while back, he wasn't sure it would be feasible due the need to check file location as well as extension. Are there any news on this front?
